course/addquestions.php

```diff
@@ -144,11 +153,31 @@
 			require_once('../includes/filehandler.php');
 			deleteallaidfiles($aid);
 			if ($aver > 1) {
+                $stm = $DBH->query("SELECT userid,score FROM imas_assessment_records WHERE assessmentid=:assessmentid");
+                while ($row = $stm->fetch(PDO::FETCH_ASSOC)) {
+                    $grades[$row['userid']]=$row["score"];
+                }
 				$stm = $DBH->prepare("DELETE FROM imas_assessment_records WHERE assessmentid=:assessmentid");
 			} else {
-				$stm = $DBH->prepare("DELETE FROM imas_assessment_sessions WHERE assessmentid=:assessmentid");
+                $query = "SELECT userid, bestscores FROM imas_assessment_sessions WHERE assessmentid=$aid";

```

Comment:
This should probably be done using `prepare` and execute, rather than sticking $aid directly into the query.